nilearn/decoding/searchlight.py

```diff
@@ -353,7 +382,58 @@ def fit(self, imgs, y, groups=None):
             self.n_jobs,
             self.verbose,
         )
+        self.process_mask_ = process_mask
+        self.masked_scores_ = scores
         scores_3D = np.zeros(process_mask.shape)
-        scores_3D[process_mask] = scores
+        scores_3D[np.where(process_mask)] = scores
         self.scores_ = scores_3D
         return self
+
+    @property
+    def scores_img_(self):
+        """Convert the 3D scores array into a NIfTI image."""
+        if self.scores_ is None:
+            raise ValueError(
+                "The model has not been fitted yet. Call `fit()` "
+                "before accessing `scores_img_`."
+            )
+        return new_img_like(self.mask_img, self.scores_)
+
+    def transform(self, imgs):
+        """Apply the fitted searchlight on new images."""
+        if self.process_mask_ is None or self.scores_ is None:
+            raise ValueError(
+                "You must fit the model before calling `transform()`."
+            )
+
+        # Ensure images are 4D
+        imgs = check_niimg_4d(imgs)
+
+        # Apply mask and affinity again with the fitted mask
+        X, _ = _apply_mask_and_get_affinity(
+            np.asarray(np.where(self.process_mask_)).T,
+            imgs,
+            self.radius,
+            True,
+            mask_img=self.mask_img,
+        )
+
+        # Ensure the result from search_light is not None and correctly shaped
+        result = search_light(
+            X,
+            None,
+            self.estimator,
+            None,
+            None,
+            self.scoring,
+            self.cv,
+            self.n_jobs,
+            self.verbose,
+        )
+        if result is None or result.size == 0:
```

Comment:
Wondering when this is supposed to happen ?

Reply:
so should I remove this?

Reply:
I think so, if there is no explicit reason to keep it.